src/eterna/pose2D/Pose2D.ts

```diff
@@ -66,11 +66,16 @@ export const PLAYER_MARKER_LAYER = 'Markers';
 export const SCRIPT_MARKER_LAYER = 'Script';
 
 export type PoseMouseDownCallback = (e: InteractionEvent, closestDist: number, closestIndex: number) => void;
+export type PosePickCallback = (closestIndex: number) => void;
 
 export default class Pose2D extends ContainerObject implements Updatable {
     public static readonly COLOR_CURSOR: number = 0xFFC0CB;
     public static readonly ZOOM_SPACINGS: number[] = [45, 30, 20, 14, 7];
 
+    public readonly baseMarked = new Signal<number>();
```

Comment:
You mean using signals instead of setting callbacks? I've always been unhappy with the current approach using maybe-defined callbacks - the whole point of signals is to have dynamically registered callbacks in a less fragile way.

Reply:
well, having this member at all -- we managed to have base marking mechanisms in Pose2D for some time without it, so I don't quite get what it adds. do we need the single most recent marked base for a reason?

Reply:
Ah! Yeah. This is so that we can propagate 2D base marks to the 3D view

Reply:
totally makes sense then; i didn't catch the code that achieved that on the 3d side